### PR TITLE
[4.0] [com_config] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
use prepared statement for SQL

### Testing Instructions
test com_config

### Expected result
should work as before

### Actual result
N/A

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 849cd392bd958c4d2cbd84130567c93be9517004<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25193">issues.joomla.org/tracker/joomla-cms/25193</a>.</sub>

File: administrator/components/com_config/Model/ApplicationModel.php [hunk 1]
```diff
@@ -826,7 +827,8 @@ public function storePermissions($permission = null)
 			$query = $this->getDbo()->getQuery(true)
```
Comment (SharkyKZ) [inline administrator/components/com_config/Model/ApplicationModel.php]:
```suggestion
			$query = $this->_db->getQuery(true)
```

File: administrator/components/com_config/Model/ApplicationModel.php [hunk 2]
```diff
@@ -826,7 +827,8 @@ public function storePermissions($permission = null)
...
 				->select($this->getDbo()->quoteName('id'))
```
Comment (SharkyKZ) [inline administrator/components/com_config/Model/ApplicationModel.php]:
```suggestion
				->select($this->_db->quoteName('id'))
```

File: administrator/components/com_config/Model/ApplicationModel.php [hunk 3]
```diff
@@ -826,7 +827,8 @@ public function storePermissions($permission = null)
...
 				->from($this->getDbo()->quoteName('#__assets'))
```
Comment (SharkyKZ) [inline administrator/components/com_config/Model/ApplicationModel.php]:
```suggestion
				->from($this->_db->quoteName('#__assets'))
```

File: administrator/components/com_config/Model/ApplicationModel.php [hunk 4]
```diff
@@ -826,7 +827,8 @@ public function storePermissions($permission = null)
...
-				->where($this->getDbo()->quoteName('name') . ' = ' . $this->getDbo()->quote($permission['component']));
+				->where($this->getDbo()->quoteName('name') . ' = :component')
```
Comment (SharkyKZ) [inline administrator/components/com_config/Model/ApplicationModel.php]:
```suggestion
				->where($this->_db->quoteName('name') . ' = :component')
```

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on afe9cddc2a790361085b5e2527c82430ce1f0bf6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25193">issues.joomla.org/tracker/joomla-cms/25193</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on afe9cddc2a790361085b5e2527c82430ce1f0bf6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25193">issues.joomla.org/tracker/joomla-cms/25193</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25193">issues.joomla.org/tracker/joomla-cms/25193</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**